src/libraries/Hooks.sol

```diff
@@ -161,12 +168,22 @@ library Hooks {
         IPoolManager.ModifyPositionParams memory params,
         bytes calldata hookData
     ) internal returns (bool shouldExecute) {
-        if (self.hasPermission(BEFORE_MODIFY_POSITION_FLAG)) {
-            shouldExecute = self.callHookNoopable(
-                abi.encodeWithSelector(IHooks.beforeModifyPosition.selector, msg.sender, key, params, hookData)
-            );
+        if (params.liquidityDelta >= 0) {
+            if (key.hooks.hasPermission(BEFORE_ADD_LIQUIDITY_FLAG)) {
+                shouldExecute = self.callHookNoopable(
+                    abi.encodeWithSelector(IHooks.beforeAddLiquidity.selector, msg.sender, key, params, hookData)
+                );
+            } else {
+                shouldExecute = true;
+            }
         } else {
-            return true;
+            if (key.hooks.hasPermission(BEFORE_REMOVE_LIQUIDITY_FLAG)) {
+                shouldExecute = self.callHookNoopable(
+                    abi.encodeWithSelector(IHooks.beforeRemoveLiquidity.selector, msg.sender, key, params, hookData)
+                );
+            } else {
+                shouldExecute = true;
+            }
```

Comment:
Feel like this looks a little ugly, could also change to something like:
```
    /// @notice calls beforeModifyPosition hook if permissioned and validates return value
    function beforeModifyPosition(
        IHooks self,
        PoolKey memory key,
        IPoolManager.ModifyPositionParams memory params,
        bytes calldata hookData
    ) internal returns (bool shouldExecute) {
        if (params.liquidityDelta >= 0 && key.hooks.hasPermission(BEFORE_ADD_LIQUIDITY_FLAG)) {
            shouldExecute = self.callHookNoopable(
                abi.encodeWithSelector(IHooks.beforeAddLiquidity.selector, msg.sender, key, params, hookData)
            );
        } else if (params.liquidityDelta < 0 && key.hooks.hasPermission(BEFORE_REMOVE_LIQUIDITY_FLAG)) {
            shouldExecute = self.callHookNoopable(
                abi.encodeWithSelector(IHooks.beforeRemoveLiquidity.selector, msg.sender, key, params, hookData)
            );
        } else {
            shouldExecute = true;
        }
    }
 ```
It's a little more gas ~100 but maybe that's ok?

Reply:
I think we should expose separate getters for querying beforeAdd/RemoveLiquidity and deprecating beforeModifiyPosition + afterModifyPosition so you don't need this conditional

Reply:
I think I prefer this branching. And @zhongeric I think Mark and I talked about this and we kinda liked having the beforeModifyPos/afterModifyPos as the entrypoints in the Hooks library so that we are removing the liquidityDelta logic from core codepaths